projects/common/src/time/user-specified-time-range.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { isNil } from 'lodash-es';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged, map, share } from 'rxjs/operators';
+import { NavigationService } from '../navigation/navigation.service';
+import { PreferenceService, StorageType } from '../preference/preference.service';
+import { TimeRange } from './time-range';
+import { TimeRangeService } from './time-range.service';
+
+@Injectable({ providedIn: 'root' })
+export class UserSpecifiedTimeRangeService {
+  private static readonly STORAGE_TYPE: StorageType = StorageType.Session;
+  private static readonly TIME_RANGE_PREFERENCE_KEY: string = 'page-time-range';
+
+  private readonly pageTimeRangesSubject$: BehaviorSubject<PageTimeRangeMap> = new BehaviorSubject<PageTimeRangeMap>(
+    {}
+  );
+  private readonly storedTimeRanges$: Observable<PageTimeRangeMap> = this.preferenceService.get<PageTimeRangeMap>(
+    UserSpecifiedTimeRangeService.TIME_RANGE_PREFERENCE_KEY,
+    {},
+    UserSpecifiedTimeRangeService.STORAGE_TYPE
+  );
+
+  public constructor(
+    private readonly preferenceService: PreferenceService,
+    private readonly timeRangeService: TimeRangeService,
+    private readonly navigationService: NavigationService
+  ) {
+    this.storedTimeRanges$.subscribe(values => {
+      this.pageTimeRangesSubject$.next(values);
+    });
+  }
+  private getDefaultPageTimeRange(path: string): TimeRange {
+    const defaultTimeRange = this.navigationService.getRouteConfig([path], this.navigationService.rootRoute())?.data
+      ?.defaultTimeRange;
+
+    if (!defaultTimeRange) {
+      //  Use current time range as default default
+      return this.timeRangeService.getCurrentTimeRange();
+    }
+
+    return defaultTimeRange;
+  }
+
+  public getUserSpecifiedTimeRangeForPage(path: string): Observable<TimeRange> {
```

Comment:
aah. I think he had that name in the beginning but I asked him to change it. I suggested UserSpecified because users can actually override the default time range (and PageTimeRangeService sounded confusing with TimeRangeService). 

In a way this service is temp. Eventually, we can merge it with the TR service when we remove the FF. For now, it is safer/cleaner for HT-ui to keep them separate.

Reply:
UserSpecifiedTimeRange is **way** more confusing than PageTimeRange, just saying. The two functional changes are that we're holding a selection across sessions, and we're picking specific time ranges for each page. But users have always been able to specify a time range, that's not unique to this service.

What's the blocker for merging this with TR service? If we want to keep it tied to the nav list, I could buy that (with a name like NavItemTimeRange), as long as that from a consumer point of view time ranges work the exact same way as always - both set and read from existing TR service.